Assets/Features/Building/Scripts/Grid/Grid.cs

```diff
@@ -220,19 +175,16 @@ public bool BuildTile(Vector3Int gridVector, float speed)
                     _cells[tile.x, tile.y, tile.z].WorkLoad);
 
                 CellData cellData = _cells[tile.x, tile.y, tile.z];
-
-                float buildAmount = _buildingStaringOpacity + (cellData.CurrentWorkLoad / cellData.WorkLoad * (1 - _buildingStaringOpacity));
-
-                Color color = _tilemap.GetColor(tile);
-                color.a = buildAmount;
-                _gridColorBuffer.Add(new TileColorData()
+                
+                GameManager.Instance.EventManager.TriggerEvent(EventId.OnChangeColorTile, new TileColorData()
```

Comment:
It's difficult to check, but it seems like you use an event to trigger another script and continue changing the color. Is this true? Since the Grid and GridTileMap are in the same functionality, we tend to not use events to communicate within the same functionality